### PR TITLE
WIP: table info process

### PR DESCRIPTION
**Pull request by jychoi-hpc**
WIP. No merge yet. 

This is to process the table info (from XGC) to calculate dr/En_dr and do plotting.

### CODE REVIEW

**Review by seunghoeku — APPROVED** [review]

File: diffusion/diffusion.py [hunk 1]
```diff
@@ -153,25 +163,38 @@ def read_stats(self, inds=Ellipsis):
...
+                i_dr_std = sqrt(i_dr_squared_average - i_dr_average**2)
+                e_dr_std = sqrt(e_dr_squared_average - e_dr_average**2)
+                i_dE_std = sqrt(i_dE_squared_average - i_dE_average**2)
+                e_dE_std = sqrt(e_dE_squared_average - e_dE_average**2)
+
+                dr_std[itri] += i_dr_std + e_dr_std
+                En_dr_std[itri] += i_dE_std + e_dE_std
+                dr_avg[itri] += i_dr_average + e_dr_average
+                En_dr_avg[itri] += i_dE_average + e_dE_average
+                marker_den[itri] += i_marker_den + e_marker_den
```
Comment (rmchurch) [inline diffusion/diffusion.py:189]: We don't want to add the separate ion and electrons here, but keep ions and electrons separate. I didn't have the code setup to handle ions and electrons previously, but what we will want is separate graphs of these

File: diffusion/diffusion.py [hunk 2]
```diff
@@ -153,25 +163,38 @@ def read_stats(self, inds=Ellipsis):
...
+                e_dr_squared_average = row[7]
+                e_dE_average = row[8]
+                e_dE_squared_average = row[9]
+                e_marker_den = row[10]
+
+                i_dr_std = sqrt(i_dr_squared_average - i_dr_average**2)
+                e_dr_std = sqrt(e_dr_squared_average - e_dr_average**2)
+                i_dE_std = sqrt(i_dE_squared_average - i_dE_average**2)
+                e_dE_std = sqrt(e_dE_squared_average - e_dE_average**2)
+
```
Comment (rmchurch) [inline diffusion/diffusion.py:184]: I don't think you want to do this `sqrt` part here. This is the function of `calc_sigma`, and so you can see this currently misses the `1./marker_den factor`. Essentially what we need to do is

File: diffusion/diffusion.py (hunk 1 above)
Comment (rmchurch) [reply diffusion/diffusion.py:189]: You can delete this section and I can come through and update the code for ion and electrons

File: diffusion/diffusion.py (hunk 2 above)
Comment (rmchurch) [reply diffusion/diffusion.py:184]: Essentially what we need to do is add the individual `i_dr_squared_average`, `i_marker_den`, etc. to themselves, to gather from all the processes, then we can use `calc_sigma` later on when aggregating

**Comment by jychoi-hpc** [comment]
Thanks for the comments. I understand now. I am dropping this PR but will create another one.

**Pull request closed without merging**